pkg/services/scan.go

```diff
@@ -0,0 +1,106 @@
+package services
+
+import (
+	"fmt"
+	"github.com/kptm-tools/common/common/events"
+	"github.com/kptm-tools/core-service/pkg/domain"
+	"github.com/kptm-tools/core-service/pkg/interfaces"
+	"sync"
+)
+
+type ScanService struct {
+	storage interfaces.IStorage
+}
+
+var _ interfaces.IScanService = (*ScanService)(nil)
+
+func NewScanService(storage interfaces.IStorage) *ScanService {
+	return &ScanService{
+		storage: storage,
+	}
+}
+
+type resultChannel struct {
+	data domain.Host
+	err  string // you must use error type here
+}
+
+func (s ScanService) CreateScans(hostIDs []string) (*domain.Scan, error) {
+	scanDB := domain.NewScan()
+
+	var wg sync.WaitGroup
+	var results []resultChannel
+	ch := make(chan resultChannel)
```

Comment:
To improve concurrency and prevent synchronous blocking, you can use a buffered channel here.

```suggestion
	ch := make(chan resultChannel, len(hostIDs)) // buffered channel
```

Reply:
Thanks teacher, I did not know that